site/_docs_1_0/httpproxy.md

```diff
@@ -990,6 +990,31 @@ spec:
       weight: 20
 ```
 
+## Upstream Validation
+
+When defining upstream services on a route, it's possible to configure the connection from Envoy to the backend endpoint to communicate over TLS.
+A CA certificate can be supplied to verify the backend endpoint contains a valid certificate.
+A `SubjectName` can also be supplied to verify the backend endpoint in addition to the CA cert.
+
+- The secret referenced must be type "Opaque" where the key of the secret is named: `ca.crt`
+- The service must be annotated with a Contour specific annotation: `projectcontour.io/upstream-protocol.tls: <port>` ([see annotations section](annotations.md))
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: blog
+  namespace: marketing
+spec:
+  routes:
+    - services:
+        - name: s2
+          port: 80
+          validation:
+            caSecret: foo-ca-cert
```

Comment:
Is it possible to specify a secret that is in a different namespace?

Reply:
No, the secret must be in the same namespace as the service. It would be interesting to allow this to work with cert delegation (or it might already, haven't tested that scenario out)